File: Archive/client/src/renderer/src/types/local/conversation.ts
/**
 * Private Mode - 本地会话类型
 */

/**
 * 本地会话类型
 */
export type LocalConversation = {
  id: string
  title: string
  createdAt: Date
  updatedAt: Date
}

/**
 * 数据库会话类型（从 IPC 接收）
 */
export type DBLocalConversation = {
  id: string
  title: string
  createdAt: Date | number
  updatedAt: Date | number
}

/**
 * 转换数据库会话为应用层会话
 */
export function dbConversationToLocal(conversation: DBLocalConversation): LocalConversation {
  return {
    id: conversation.id,
    title: conversation.title,
    createdAt: conversation.createdAt instanceof Date
      ? conversation.createdAt
      : new Date(conversation.createdAt),
    updatedAt: conversation.updatedAt instanceof Date
      ? conversation.updatedAt
      : new Date(conversation.updatedAt),
  }
}

File: Klee/Model/Conversation.swift
//
//  Conversation.swift
//  Klee
//
//  Data model for a single conversation (chat session).
//  Each conversation is persisted as a separate JSON file.
//

import Foundation

/// A single conversation containing a list of messages
struct Conversation: Identifiable, Codable, Equatable {
    let id: UUID
    var title: String
    var messages: [ChatMessage]
    var inspectorItems: [InspectorItem]
    let createdAt: Date
    var updatedAt: Date

    /// Whether the title is still the default placeholder
    var hasDefaultTitle: Bool {
        title == Conversation.defaultTitle
    }

    static let defaultTitle = "New Task"

    init(id: UUID = UUID(), title: String = Conversation.defaultTitle, messages: [ChatMessage] = [], inspectorItems: [InspectorItem] = [], createdAt: Date = Date(), updatedAt: Date = Date()) {
        self.id = id
        self.title = title
        self.messages = messages
        self.inspectorItems = inspectorItems
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }

    /// Custom decoder for backward compatibility with existing JSON files that lack inspectorItems
    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        id = try container.decode(UUID.self, forKey: .id)
        title = try container.decode(String.self, forKey: .title)
        messages = try container.decode([ChatMessage].self, forKey: .messages)
        inspectorItems = try container.decodeIfPresent([InspectorItem].self, forKey: .inspectorItems) ?? []
        createdAt = try container.decode(Date.self, forKey: .createdAt)
        updatedAt = try container.decode(Date.self, forKey: .updatedAt)
    }
}
